### PR TITLE
Cleanup NativeProxy on Android

### PR DESCRIPTION
**Pull request by jwajgelt**

### CODE REVIEW

File: android/src/main/cpp/NativeProxy.cpp [hunk 1]
```diff
@@ -160,43 +140,22 @@ void NativeProxy::installJSIBindings(
         jni::make_jstring(propName.utf8(rt).c_str());
     auto result = method(javaPart_.get(), viewTag, propNameJStr);
     std::string str = result->toStdString();
-    return jsi::Value(rt, jsi::String::createFromAscii(rt, str.c_str()));
+    return {rt, jsi::String::createFromAscii(rt, str)};
```
Comment (tomekzaw) [inline android/src/main/cpp/NativeProxy.cpp:143]: plz don't

File: android/src/main/cpp/NativeProxy.cpp [hunk 2]
```diff
@@ -334,10 +252,16 @@ void NativeProxy::installJSIBindings(
 
   std::weak_ptr<ErrorHandler> weakErrorHandler = errorHandler;
 
-  layoutAnimations->cthis()->setAnimationStartingBlock(
-      [wrt, weakModule, weakErrorHandler](
+  layoutAnimations_->cthis()->setAnimationStartingBlock(
+      [weakModule](
           int tag, int type, alias_ref<JMap<jstring, jstring>> values) {
-        auto &rt = *wrt.lock();
+        auto reaModule = weakModule.lock();
```
Comment (tomekzaw) [inline android/src/main/cpp/NativeProxy.cpp:258]:
```suggestion
        auto module = weakModule.lock();
```

File: android/src/main/cpp/NativeProxy.cpp [hunk 3]
```diff
@@ -334,10 +252,16 @@ void NativeProxy::installJSIBindings(
...
+        if (reaModule == nullptr) {
+          return;
+        }
+        auto &rt = *reaModule->runtime;
```
Comment (tomekzaw) [inline android/src/main/cpp/NativeProxy.cpp:262]:
```suggestion
        jsi::Runtime &rt = *reaModule->runtime;
```

File: android/src/main/cpp/NativeProxy.cpp (hunk 1 above)
Comment (jwajgelt) [reply android/src/main/cpp/NativeProxy.cpp]: honest mistake, will change this

File: android/src/main/cpp/NativeProxy.cpp [hunk 4]
```diff
@@ -584,9 +324,233 @@ int NativeProxy::subscribeForKeyboardEvents(
...
+}
+
+void NativeProxy::setupLayoutAnimations() {
+  auto weakModule =
+      std::weak_ptr<NativeReanimatedModule>(nativeReanimatedModule_);
+
+  layoutAnimations_->cthis()->setAnimationStartingBlock(
+      [weakModule](
+          int tag, int type, alias_ref<JMap<jstring, jstring>> values) {
+        auto reaModule = weakModule.lock();
```
Comment (tomekzaw) [inline android/src/main/cpp/NativeProxy.cpp:479]: I know that `module` will be a keyword in C++20 but it's also used in various many places so for now let's be consistent and use `module` here and some time in the future we will rename it everywhere.
```suggestion
        auto module = weakModule.lock();
```

Comment (jwajgelt) [reply android/src/main/cpp/NativeProxy.cpp]: I'm pretty sure C++20 shouldn't break `module` being used as an identifier anyway, but let's not get into that. 😛

**Review by piaskowyk — APPROVED** [review]

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**